src/app/ui/editor/moving_slice_state.h

```diff
@@ -25,26 +31,213 @@ namespace app {
+        totalBounds.nineSlice(m_item->oldKey.center(), bounds);
+        for (int i=0; i<9; ++i) {
+          if (!bounds[i].isEmpty()) {
+            ImageRef img;
+            img.reset(Image::create(src->pixelFormat(), bounds[i].w, bounds[i].h));
+            img->copy(src, gfx::Clip(0, 0, bounds[i]));
+            m_part[i]->copy(img.get());
+          }
+        }
+      }
+
+    private:
+      std::unique_ptr<SingleSlice> m_part[9] = {nullptr, nullptr, nullptr,
+                                                nullptr, nullptr, nullptr,
+                                                nullptr, nullptr, nullptr};
+    };
+
+
     struct Item {
       doc::Slice* slice;
       doc::SliceKey oldKey;
       doc::SliceKey newKey;
+      // Vector of each selected layer's part of the sprite under
+      // the slice bounds that will be transformed when Slice Transform is
+      // enabled. Contains one ItemContentRef by layer.
+      std::vector<ItemContentRef> content;
+      // Part of the sprite of each selected layer's merged into one image per
+      // slice. This is used to give feedback to the users when they are
+      // transforming the selected slices.
+      ItemContentRef mergedContent;
+
+      // Adds image to the content vector. The image should correspond to some
+      // part of a single layer cel's image.
+      // Internally this method builds a merged version of the images to speed
+      // up the drawing when the user updates this Item's slice in real time.
+      void pushContent(const ImageRef& image) {
+        if (content.empty()) {
+          const gfx::Rect& srcBounds = image->bounds();
+          ImageRef mergedImage;
+          mergedImage.reset(Image::create(image->pixelFormat(), srcBounds.w, srcBounds.h));
+          mergedImage->clear(image->maskColor());
+          mergedContent = (this->oldKey.hasCenter() ? (ItemContentRef)std::make_shared<NineSlice>(this, mergedImage)
+                                                    : std::make_shared<SingleSlice>(this, mergedImage));
+        }
+
+        mergedContent->copy(image.get());
+
+        ItemContentRef ssc = (this->oldKey.hasCenter() ? (ItemContentRef)std::make_shared<NineSlice>(this, image)
+                                                       : std::make_shared<SingleSlice>(this, image));
+        content.push_back(ssc);
+      }
+
+      bool isNineSlice() const { return oldKey.hasCenter(); }
+
+      // If this item manages a 9-slice key, returns the border surrounding the
+      // center rectangle of the SliceKey. Otherwise returns an empty border.
+      gfx::Border border() const {
+        gfx::Border border;
+        if (isNineSlice()) {
+          border = gfx::Border(oldKey.center().x,
+                               oldKey.center().y,
+                               oldKey.bounds().w-oldKey.center().x2(),
+                               oldKey.bounds().h-oldKey.center().y2());
+        }
+        return border;
+      }
```

Comment:
I think I could refactor this into a SliceKey gfx::Border border() method. Let me know if it makes sense and I'll change this.